node_helper.js

```diff
@@ -58,93 +58,98 @@ module.exports = NodeHelper.create({
             //Get devices first
             this.getDevicesAsync(this.config, this.pusher).then(function (response) {
                 //Load pushes on start if we need to
-                if (self.config.showPushesOnLoad) {                
-                    self.loadPushes(self.pusher, self.config, true);                
+                if (self.config.showPushesOnLoad) {
+                    self.loadPushes(self.pusher, self.config, true);
                 }
-            });                
+            });
 		}
 	},
 
     //Start listening to pushbullet
 	pushBulletListener: function(config){
 		this.pusher = new PushBullet(config.accessToken); // PushBulletAPI object
-		var stream = this.pusher.stream();
-		var self = this;
-
-        //Connect
-		stream.on('connect', function() {
-			// stream has connected
-            self.info("PushBullet connected");
-            self.connected = true;
-		});
-
-        //Logs errors
-        stream.on('error', function (error) {
-            self.error("Push stream error: "+error);                
-        });
-
-        //Watch for normal pushes if configured
-        if (config.showPushes) {
-            /*PushBullet API
-             * https://docs.pushbullet.com/#realtime-event-stream
-             * 
-             * Watch for normal pushes not Ephemerals. Tickle means something has changed on server, subtype tells what has changed.
-             */
-            stream.on("tickle", function (type) {
-                self.debug("tickle received: " + type);
-                if (type === "push") {
-                    self.loadPushes(self.pusher, config, false);
-                }
+        const self = this;
+        this.pusher.me(function(error, user) {
+            // needed to call me() to gather user iden
+            self.pusher.enableEncryption(config.endToEndPassword, user.iden);
```

Comment:
What happens if no password is set in the config and config.endToEndPassword is null?